scripts/onboarding/aks/onboarding-msi-terraform-syslog/main.tf

```diff
@@ -0,0 +1,70 @@
+resource "azurerm_resource_group" "rg" {
+  name     = var.aks_resource_group_name
+  location = var.resource_group_location
+}
+
+resource "azurerm_kubernetes_cluster" "k8s" {
+  name                = var.cluster_name
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  dns_prefix          = var.dns_prefix
+
+  tags = var.resource_tag_values
+  
+  default_node_pool {
+    name       = "agentpool"
+    vm_size    = var.vm_size
+    node_count = var.agent_count
+  }
+
+  identity {
+    type = var.identity_type
+  }
+
+  oms_agent {
+    log_analytics_workspace_id = var.workspace_resource_id
+    msi_auth_for_monitoring_enabled = true
+  }
+}
+
+resource "azurerm_monitor_data_collection_rule" "dcr" {
+  name                = "MSCI-${var.workspace_region}-${var.cluster_name}"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+
+  destinations {
+    log_analytics {
+      workspace_resource_id = var.workspace_resource_id
+      name                  = "ciworkspace"
+    }
+  }
+
+  data_flow {
+    streams      = ["Microsoft-ContainerInsights-Group-Default","Microsoft-Syslog"]
+    destinations = ["ciworkspace"]
+  }
+
+  data_sources {
+    syslog{
+      streams            = ["Microsoft-Syslog"]
+      facility_names      = var.syslog_facilities
+      log_levels          = var.syslog_levels
+      name               = "sysLogsDataSource"
+    }
+
+    extension {
+      streams            = ["Microsoft-ContainerInsights-Group-Default"]
+      extension_name     = "ContainerInsights"
+      name = "ContainerInsightsExtension"
+    }
+  }
+
+  description = "DCR for Azure Monitor Container Insights"
+}
+
+resource "azurerm_monitor_data_collection_rule_association" "dcra" {
```

Comment:
where dependencies between DCR, DCR-A and Monitoring addon defined? we would need DCR, AKS cluster and DCR-A, and monitoring addon so that data starts flowing immediately.